pkg/upgrade/control_plane.go

```diff
@@ -272,12 +273,17 @@ func (u *ControlPlaneUpgrader) updateMachines(machines *clusterapiv1alpha1.Machi
 			continue
 		}
 
-		originalProviderID, err := kubernetes.ParseProviderID(*machine.Spec.ProviderID)
+		originalProviderID, err := noderefutil.NewProviderID(*machine.Spec.ProviderID)
 		if err != nil {
 			return err
 		}
 
-		oldNode := u.GetNodeFromProviderID(originalProviderID)
+		oldNode := u.GetNodeFromProviderID(originalProviderID.ID())
+		if oldNode == nil {
+			u.log.Info("Couldn't retrieve oldNode", "id", fmt.Sprintf("%q", originalProviderID))
```

Comment:
Do you need `fmt.Sprintf` here?

Reply:
probably not, I was debugging whitespace issues.